sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1657,21 +1657,12 @@ class Analyzer(
 
       // Third, for every Window Spec, we add a Window operator and set currentChild as the
       // child of it.
-      var currentChild = child
-      var i = 0
-      while (i < groupedWindowExpressions.size) {
-        val ((partitionSpec, orderSpec), windowExpressions) = groupedWindowExpressions(i)
-        // Set currentChild to the newly created Window operator.
-        currentChild =
-          Window(
-            windowExpressions,
-            partitionSpec,
-            orderSpec,
-            currentChild)
-
-        // Move to next Window Spec.
-        i += 1
-      }
+      val currentChild =
+        groupedWindowExpressions.foldLeft(child) {
+          case (last, ((partitionSpec, orderSpec), windowExpressions)) =>
+            // Set currentChild to the newly created Window operator.
```

Comment:
Sorry, one last thing, can you remove this comment? and update the comment above and below this expression? they're no longer applicable. Actually `currentChild` isn't a great name now anyway. eh, `windowOps` or something?